packages/graphql/src/schema/validation/custom-rules/validate-authorization-like-directives.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ASTVisitor, DirectiveNode } from "graphql";
+import { extendSchema } from "graphql";
+import type { SDLValidationContext } from "graphql/validation/ValidationContext";
+import { VALIDATION_ERROR_CODES } from "../utils/validation-error-codes";
+import { createGraphQLError } from "./utils/document-validation-error";
+import { getPathToNode } from "./utils/path-parser";
+import { assertArgumentType, findArgumentDefinitionNodeByName } from "./utils/utils";
+
+// ValidateAuthorizationLikeDirectives validates the directives subscriptionsAuthorization, authorization, authentication
```

Comment:
This is probably more useful as a tsdoc comment:

```suggestion
/** validates the directives subscriptionsAuthorization, authorization, authentication **/
```